### PR TITLE
[Merged by Bors] - perf: increase priority of Algebra.id

### PR DESCRIPTION
**Pull request by MichaelStollBayreuth**
Inspired by [this discussion](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/erw.20and.20IsDedekindDomain.2EHeightOneSpectrum.2EadicCompletion/near/439380909), we increase the priority of the `Algebra.id` instance to see if this has a positive effect on instance synthesis speed.

Originally, this led to two proofs breaking, but (thanks to @kbuzzard) this can be fixed easily without the need to restore the default instance priority.

The net effect is that this saves about 0.1% of build instructions.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by MichaelStollBayreuth** [comment]
!bench

**Comment by MichaelStollBayreuth** [comment]
!bench

File: Mathlib/RingTheory/Jacobson.lean [hunk 1]
```diff
@@ -514,6 +514,7 @@ theorem isMaximal_comap_C_of_isMaximal [Nontrivial R] (hP' : ∀ x : R, C x ∈
 set_option linter.uppercaseLean3 false in
 #align ideal.polynomial.is_maximal_comap_C_of_is_maximal Ideal.Polynomial.isMaximal_comap_C_of_isMaximal
 
+attribute [instance 1000] Algebra.id in
```
Comment (kbuzzard) [inline Mathlib/RingTheory/Jacobson.lean:517]: `  refine' RingHom.IsIntegral.tower_bot φ (algebraMap (R[X] ⧸ P) (Localization M')) _ _` fixes this one.

**Comment by MichaelStollBayreuth** [comment]
@kbuzzard Feel free to push these changes!

**Comment by Ruben-VandeVelde** [comment]
!bench

**Comment by MichaelStollBayreuth** [comment]
build instructions:    -129.706 * 10⁹

Files with an instruction count that changed by at least 10⁹:

Mathlib.LinearAlgebra.TensorProduct.Graded.External	instructions		+7.075 * 10⁹
Mathlib.Algebra.DualQuaternion	instructions	+2.736 * 10⁹
Mathlib.Analysis.NormedSpace.Star.Matrix	instructions		+1.178 * 10⁹
Mathlib.Algebra.Lie.Weights.Killing	instructions	+1.056 * 10⁹

Mathlib.RingTheory.Jacobson	instructions		-10.429 * 10⁹
Mathlib.MeasureTheory.Measure.FiniteMeasure	instructions		-7.82 * 10⁹
Mathlib.MeasureTheory.Decomposition.Lebesgue	instructions		-6.485 * 10⁹
Mathlib.AlgebraicGeometry.StructureSheaf	instructions		-5.945 * 10⁹
Mathlib.Topology.ContinuousFunction.StoneWeierstrass	instructions		-4.61 * 10⁹
Mathlib.Analysis.Seminorm	instructions		-4.464 * 10⁹
Mathlib.RepresentationTheory.GroupCohomology.Resolution	instructions		-3.739 * 10⁹
Mathlib.MeasureTheory.Measure.Haar.Unique	instructions		-3.159 * 10⁹
Mathlib.Analysis.NormedSpace.QuaternionExponential	instructions		-3.11 * 10⁹ 
Mathlib.Analysis.LocallyConvex.WithSeminorms	instructions		-2,873 * 10⁹ 
Mathlib.MeasureTheory.Measure.ProbabilityMeasure	instructions		-2.461 * 10⁹
Mathlib.RingTheory.IntegralClosure	instructions		-2.415 * 20⁹
Mathlib.MeasureTheory.Covering.Differentiation	instructions		-2.394 * 10⁹
Mathlib.Analysis.NormedSpace.Star.ContinuousFunctionalCalculus.Instances	instructions		-2,042 * 10⁹
Mathlib.NumberTheory.Cyclotomic.Basic	instructions		-1.977 * 10⁹
Mathlib.Analysis.NormedSpace.Spectrum	instructions		-1.938 * 10⁹
Mathlib.Topology.ContinuousFunction.UniqueCFC	instructions		-1.921 * 10⁹
Mathlib.Analysis.Calculus.MeanValue	instructions		-1.889 * 10⁹ 
Mathlib.MeasureTheory.Measure.FiniteMeasureProd	instructions		-1.728 * 10⁹
Mathlib.Analysis.Convex.Strong	instructions	-1.727 * 10⁹
Mathlib.MeasureTheory.Function.L1Space	instructions		-1.46 * 10⁹
Mathlib.NumberTheory.NumberField.CanonicalEmbedding.ConvexBody	instructions		-1.351 * 10⁹
Mathlib.NumberTheory.NumberField.ClassNumber	instructions		-1.234 * 10⁹
Mathlib.MeasureTheory.Function.LpSeminorm.Basic	instructions		-1.231 * 10⁹
Mathlib.CategoryTheory.Category.Cat.Limit	instructions		-1.219 * 10⁹
Mathlib.RingTheory.Kaehler	instructions		-1.199 * 10⁹
Mathlib.MeasureTheory.Measure.Hausdorff	instructions		-1.194 * 10⁹ 
Mathlib.Analysis.Calculus.Rademacher	instructions		-1.191 * 10⁹
Mathlib.Analysis.NormedSpace.Star.ContinuousFunctionalCalculus	instructions		-1.171 * 10⁹ 
Mathlib.Analysis.Convex.Integral	instructions		-1.168 * 10⁹
Mathlib.Analysis.Convex.SpecificFunctions.Basic	instructions		-1.116 * 10⁹
Mathlib.Algebra.Module.Zlattice.Basic	instructions		-1.108 * 10⁹
Mathlib.Analysis.Quaternion	instructions		-1.108 * 10⁹
Mathlib.Analysis.Convex.SpecificFunctions.Pow	instructions		-1.103 * 10⁹ 
Mathlib.Analysis.SpecialFunctions.Gamma.BohrMollerup	instructions		-1.089 * 10⁹
Mathlib.Analysis.SpecialFunctions.Bernstein	instructions		-1.074 * 10⁹
Mathlib.MeasureTheory.Measure.Lebesgue.Basic	instructions		-1.056 * 10⁹
Mathlib.NumberTheory.ClassNumber.FunctionField	instructions		-1.004 * 10⁹

**Comment by kbuzzard** [comment]
Another PR fixed the Jacobson file issue (by changing `refine'` to `refine`) so this PR is looking even more appealing now. However I wrote some of it so I'll just do

maintainer merge

**Comment by mattrobball** [comment]
bors merge

**Pull request closed without merging**